cumulus/polkadot-parachain/src/service.rs

```diff
@@ -526,403 +557,379 @@ where
+				code_hash_provider: move |block_hash| {
+					client.code_at(block_hash).ok().map(|c| ValidationCode::from(c).hash())
+				},
+				sync_oracle,
+				keystore,
+				collator_key,
+				para_id,
+				overseer_handle,
+				relay_chain_slot_duration,
+				proposer,
+				collator_service,
+				authoring_duration: Duration::from_millis(1500),
+				reinitialize: true, /* we need to always re-initialize for asset-hub moving
+				                     * to aura */
+			};
+
+			aura::run::<Block, <AuraId as AppCrypto>::Pair, _, _, _, _, _, _, _, _, _>(params).await
+		});
+
+		let spawner = task_manager.spawn_essential_handle();
+		spawner.spawn_essential("cumulus-asset-hub-collator", None, collation_future);
+
+		Ok(())
+	}
 }
 
 /// Start an aura powered parachain node which uses the lookahead collator to support async backing.
 /// This node is basic in the sense that its runtime api doesn't include common contents such as
 /// transaction payment. Used for aura glutton.
-pub async fn start_basic_lookahead_node<Net: NetworkBackend<Block, Hash>>(
-	parachain_config: Configuration,
-	polkadot_config: Configuration,
-	collator_options: CollatorOptions,
-	para_id: ParaId,
-	hwbench: Option<sc_sysinfo::HwBench>,
-) -> sc_service::error::Result<(TaskManager, Arc<ParachainClient<FakeRuntimeApi>>)> {
-	start_node_impl::<FakeRuntimeApi, _, _, _, Net>(
-		parachain_config,
-		polkadot_config,
-		collator_options,
-		CollatorSybilResistance::Resistant, // Aura
-		para_id,
-		|_, _, _, _| Ok(RpcModule::new(())),
-		build_relay_to_aura_import_queue::<_, AuraId>,
-		start_lookahead_aura_consensus,
-		hwbench,
-	)
-	.await
+pub struct BasicLookaheadNode;
+
+impl NodeSpec for BasicLookaheadNode {
+	type RuntimeApi = FakeRuntimeApi;
+	type BuildImportQueue = BuildRelayToAuraImportQueue<Self::RuntimeApi, AuraId>;
+	type BuildRpcExtensions = BuildEmptyRpcExtensions<Self::RuntimeApi>;
+	type StartConsensus = StartLookaheadAuraConsensus<Self::RuntimeApi>;
+	const SYBIL_RESISTANCE: CollatorSybilResistance = CollatorSybilResistance::Resistant;
 }
 
 /// Start a parachain node for Rococo Contracts.
-pub async fn start_contracts_rococo_node<Net: NetworkBackend<Block, Hash>>(
-	parachain_config: Configuration,
-	polkadot_config: Configuration,
-	collator_options: CollatorOptions,
-	para_id: ParaId,
-	hwbench: Option<sc_sysinfo::HwBench>,
-) -> sc_service::error::Result<(TaskManager, Arc<ParachainClient<FakeRuntimeApi>>)> {
-	start_node_impl::<FakeRuntimeApi, _, _, _, Net>(
-		parachain_config,
-		polkadot_config,
-		collator_options,
-		CollatorSybilResistance::Resistant, // Aura
-		para_id,
-		build_contracts_rpc_extensions,
-		build_aura_import_queue,
-		start_lookahead_aura_consensus,
-		hwbench,
-	)
-	.await
+pub struct ContractsRococoNode;
+
+impl NodeSpec for ContractsRococoNode {
+	type RuntimeApi = FakeRuntimeApi;
+	type BuildImportQueue = BuildAuraImportQueue<Self::RuntimeApi>;
+	type BuildRpcExtensions = BuildContractsRpcExtensions;
```

Comment:
This can also just be the `GenericLookahead` stuff. Looking at the RPC, contracts only has the `Dev` stuff added and we can just add this everywhere as well.

Reply:
Done